tests/installation/disk_activation_iscsi.pm

```diff
@@ -0,0 +1,73 @@
+# SUSE's openQA tests
+#
+# Copyright © 2016-2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: initial installation: iSCSI client for the supportserver
+#          supportserver to provide one iSCSI target which will
+#          provide a multipathed second disk to the client.
+# Maintainer: Klaus Wagner <kgw@suse.com>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    my $self = shift;
+    my $iscsi_iqn;
+    my $iscsi_server_ip;
+
+    # Defaults: the values from the supportserver and client
+    $iscsi_iqn       = get_var('ISCSI_IQN',       "iqn.2016-02.de.openqa");
+    $iscsi_server_ip = get_var('ISCSI_SERVER_IP', "10.0.2.1");
+
+    if (get_var('WITHISCSI')) {
+        # Assumption: we are within an initial installation. Due to
+        # kernel parameter "withiscsi=1" the YaST Disk Activation screen
+        # is expected to come up.
+        assert_screen 'disk-activation-iscsi', 180;
+        send_key "alt-i";    # "Configure iSCSI Disks"
+                             # screen "iSCSI Initiator Overwiew", Tab "Service"
+        assert_screen 'iscsi-initiator-service-fs', 180;
+        send_key "alt-i";    # go to initiator name field
+        wait_still_screen(2, 10);
+        type_string "$iscsi_iqn";
+        wait_still_screen(2, 10);
+        send_key "alt-n";    # "Connected Targets" tab, empty list
+        assert_screen 'iscsi-initiator-connected-targets-none-fs';
+        send_key $cmd{add};    # go to "iSCSI Initiator Discovery" screen
+        assert_screen 'iscsi-discovery-fs';
+        send_key "alt-i";      # go to IP address field
+        wait_still_screen(2, 10);
+        type_string "$iscsi_server_ip";
+        send_key "alt-n";      # iSCSI Initiator Discovery: discovered targets list, first one is selected
+        send_key "alt-o";      # press Connect button
+        wait_still_screen(2, 10);
+        assert_screen 'iscsi-initiator-startup-and-authentication';
+        send_key "alt-s";       # startup mode. Selection list: *manual onboot automatic
+        send_key "down";        # Wanted: onboot
+        send_key $cmd{next};    # Now connect
+        assert_screen 'iscsi-client-target-connected-fs';
+        send_key $cmd{next};
+        assert_screen 'iscsi-initiator-service-fs';
+        send_key $cmd{ok};
+        assert_screen 'disk-activation-iscsi';
+        send_key $cmd{next};
+    }
+    else {
+        # As of now this case should never happen.
+        # future use cases might go here
+        record_info("Nothing to do", "WITHISCSI not set to a true value");
+    }
```

Comment:
Instead of the giant if/else please use an early return, e.g.

```perl
    return record_info("Nothing to do", "WITHISCSI not set to a true value");
    # continue as in the if-branch
```

however I would recommend to make that rather explicit by simply dying if the variable is not set, e.g.

```perl
die "WITHISCSI not set" unless get_var('WITHISCSI');
# continue as in the if-branch
```

Having an explicit test failure is much better in many cases than a silently skipped "green" test which can people falsely assume that "iscsi+multipath is properly covered because we have a green test" but in reality nothing would be done at all.

Reply:
Agreed. To be done in the upcoming amended commit. (As of now this `die`  is supposed to never trigger as without `WITHISCSI` `main_common.pm` won't invoke `disk_activation_iscsi.pm` in the first place)